src/theory/arith/operator_elim.h

```diff
@@ -91,29 +104,30 @@ class OperatorElim : public EagerProofGenerator
    *
    * This method is called both during expandDefinition and during ppRewrite.
    *
+   * @param nm Pointer to the node manager
    * @param n The node to eliminate operators from.
+   * @param lems The lemmas storing (L, k) where L is the lemma and k is the 
+   * attached skolem it is associated with.
+   * @param partialOnly Whether we are only eliminating partial operators.
+   * @param wasNonLinear Set to true if n requires a non-linear logic.
    * @return The (single step) eliminated form of n.
    */
-  Node eliminateOperators(Node n,
-                          std::vector<SkolemLemma>& lems,
-                          TConvProofGenerator* tg,
-                          bool partialOnly);
-  /** get arithmetic skolem
-   *
-   * Returns the Skolem in the above map for the given id, creating it if it
-   * does not already exist.
-   */
-  Node getArithSkolem(SkolemId asi);
+  static Node eliminateOperators(NodeManager* nm,
+                                 Node n,
+                                 std::vector<std::pair<Node, Node>>& lems,
+                                 bool partialOnly,
```

Comment:
now that you deleted the option, is this flag still necessary? It seems this method is only called with the flag as `false`

Reply:
This is a bit different: this flag indicates whether we want to eliminate *only* the partial operators, the option was to force all operators to be total.

We do call this method is partialOnly=true when checking models via expandDefinitions, the call is here: https://github.com/cvc5/cvc5/blob/main/src/theory/arith/arith_rewriter.cpp#L1251.

Reply:
I see, thanks for clarifying.